modules/aws/auth.go

```diff
@@ -153,3 +154,24 @@ type CredentialsError struct {
 func (err CredentialsError) Error() string {
 	return fmt.Sprintf("Error finding AWS credentials. Did you set the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables or configure an AWS profile? Underlying error: %v", err.UnderlyingErr)
 }
+
+func getNewSession(config *aws.Config) (*session.Session, error) {
+
+	if GlobalCustomEndpoints != nil {
```

Comment:
Instead of making it a pointer, which could lead to panics due to `nil` pointer dereferences, perhaps this could check that the `map` isn't empty?